fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -0,0 +1,292 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.starrocks.analysis.DescriptorTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.StarRocksFEMetaVersion;
+import com.starrocks.common.io.Text;
+import com.starrocks.external.iceberg.IcebergCatalog;
+import com.starrocks.external.iceberg.IcebergCatalogType;
+import com.starrocks.external.iceberg.IcebergUtil;
+import com.starrocks.thrift.TColumn;
+import com.starrocks.thrift.TIcebergTable;
+import com.starrocks.thrift.TTableDescriptor;
+import com.starrocks.thrift.TTableType;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.expressions.UnboundPredicate;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class IcebergTable extends Table {
+    private static final Logger LOG = LogManager.getLogger(IcebergTable.class);
+
+    private static final String PROPERTY_MISSING_MSG =
+            "Iceberg %s is null. Please add properties('%s'='xxx') when create table";
+    private static final String JSON_KEY_ICEBERG_DB = "database";
+    private static final String JSON_KEY_ICEBERG_TABLE = "table";
+    private static final String JSON_KEY_RESOURCE_NAME = "resource";
+    private static final String JSON_KEY_ICEBERG_PROPERTIES = "icebergProperties";
+
+    private static final String ICEBERG_CATALOG = "starrocks.catalog-type";
+    private static final String ICEBERG_METASTORE_URIS = "iceberg.catalog.hive.metastore.uris";
+    private static final String ICEBERG_DB = "database";
+    private static final String ICEBERG_TABLE = "table";
+    private static final String ICEBERG_RESOURCE = "resource";
+
+    private org.apache.iceberg.Table icebergTable; // actual iceberg table
```

Comment:
nit: 
We need to discuss the abbreviation later, org.apache.iceberg.Table icebergTable and com.starrocks.catalog.icebergTable will often be used later.
I think for the sake of naming conciseness and consistency, is it necessary to change the naming of icebergTable, such as icbTbl.
What do you think about this.